Rock/Data/Entity.cs

```diff
@@ -313,7 +313,8 @@ public virtual IEntity Clone()
 
             foreach ( var propInfo in this.GetType().GetProperties() )
             {
-                if ( !propInfo.GetGetMethod().IsVirtual || virtualPropsWhiteList.Contains(propInfo.Name) )
+                MethodInfo getMethod = propInfo.GetGetMethod();
+                if ( getMethod != null && ( !getMethod.IsVirtual || virtualPropsWhiteList.Contains(propInfo.Name) ) )
```

Comment:
Hi @Tschrock , since the virtualPropsWhiteList containing the propInfo.Name **is a valid condition of the _original_ IF block**, wouldn't this be more correctly written as:

```csharp
   if ( ( getMethod != null && !getMethod.IsVirtual ) || virtualPropsWhiteList.Contains( propInfo.Name ) )
```
...this way, even if the getMethod is null, if virtualPropsWhiteList contains the propInfo.Name, it can be added to the dictionary.  Of course, if propInfo.GetValue(...) returns null, then all bets are off again.

Reply:
The problem with that logic is that the whitelist could potentially include properties that either
    1. have a private getter
or
    2. have no getter.

In case 1, it would be providing the value of a private property (something I assume you don't want to happen since it's private to start with).

In case 2, it would cause a new exception to be thrown (See the [microsoft documentation](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.propertyinfo.getvalue?view=netframework-4.7.2#System_Reflection_PropertyInfo_GetValue_System_Object_System_Object___): `PropertyInfo.GetValue(...)` will throw an `ArgumentException` if the property's getter is not found).

If you can guarantee that the whitelist will always include valid properties with public getters, then you could get away with that, but I'd rather be more defensive about things and remove the possibility of future issues.